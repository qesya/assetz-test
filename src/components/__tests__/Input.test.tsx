/* eslint-disable testing-library/prefer-screen-queries */
import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Input from '../Input';

const props = {
	value: '1',
	onChange: jest.fn(),
}

describe('<List />', () => {

	test('should does not have input label', () => {
		const { queryByTestId } = render(
			<Input {...props} />
		);
		expect(queryByTestId('input-label')).not.toBeInTheDocument();
	});

	test('should has input label', () => {
		const { getByText } = render(
			<Input label="name" {...props} />
		);
		expect(getByText('name')).toBeInTheDocument();
	});

	test('should match snapshot', () => {
		const { asFragment } = render(
			<Input label="name" {...props} />
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
