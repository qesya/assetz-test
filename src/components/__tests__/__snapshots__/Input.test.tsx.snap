// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ jeSLVl"
  >
    <label
      data-testid="input-label"
      for="name"
    >
      name
    </label>
    <input
      class="sc-gsnTZi eiayYQ"
      id="name"
      value="1"
    />
  </div>
</DocumentFragment>
`;
